### ISSUE TITLE
initial zq read support for parquet format

### ISSUE THREAD

**Issue by alfred-landrum** [brimsec/zq#703]
We want to support detecting and reading from parquet formatted files in zq, generating zng.Record's for each row, enabling zql searches and queries. We have ideas on how to optimize columnar access in the future, but for this issue, we just want to get the data into zq.

This open source pure golang library will be useful:
https://github.com/xitongsys/parquet-go

Parquet has primitive and logical types, many map cleanly into existing ZNG types, but not all:
https://github.com/apache/parquet-format#types
https://github.com/apache/parquet-format/blob/master/LogicalTypes.md

For this issue, we should support those that map cleanly onto ZNG, and note types that will require special handling for future work.

**Comment by aswan** [comment]
Initial support merged in #736 and #754

**Comment by philrz** [comment]
Verified in `zq` commit `64d5e4a`. I've attached a few sample Parquet files that can be read successfully with `zq`. They're only GZIP-compressed to make GitHub happy.

[conn.parquet.gz](https://github.com/brimsec/zq/files/4623606/conn.parquet.gz)
[dns.parquet.gz](https://github.com/brimsec/zq/files/4623607/dns.parquet.gz)
[http.parquet.gz](https://github.com/brimsec/zq/files/4623611/http.parquet.gz)

```
$ zq -t -i parquet "count() by _path" http.parquet conn.parquet dns.parquet
#0:record[_path:string,count:uint64]
0:[dns;10;]
0:[conn;10;]
0:[http;4;]
```

Some things to note:

1. `zq help` doesn't yet mention Parquet as among the supported input formats, but we'll address this (#769)
2. We know there's variations of Parquet format `zq` cannot yet handle. #764 cites one, and @mikesbrown is collecting up more examples. We'll make a plan for addressing these as time/demand allow.
3. It's expected that reading a GZIP-compressed Parquet file doesn't work, since as @aswan explained to me:
> The Parquet reader needs to do all sorts of random access. But the parquet format has all sorts of compression bells and whistles

```
# zq -t -i parquet "count() by _path" http.parquet.gz conn.parquet.gz dns.parquet.gz 
http.parquet.gz: seek http.parquet.gz: invalid argument
```

Thanks @aswan!
